enumerables.rb

```diff
@@ -0,0 +1,107 @@
+# Enumerable methods
+
+module Enumerable
+  # my_each
+
+  def my_each
+    length.times { |i| yield(self[i]) }
+    self
+  end
+
+  # end of my_each
+
+  # my_each_with_index
+  def my_each_with_index
+    length.times { |i| yield(self[i], i) }
+    self
+  end
+
+  # end of my_each_with_index
+
+  # my_select
+
+  def my_select
+    arr = []
+    my_each { |i| arr.push(i) if yield(i) }
+    arr
+  end
+
+  # end of my_select
+
+  # my_all?
+
+  def my_all?
+    if_all = true
+    my_each do |i|
+      break if_all = false unless yield(i)
+    end
+    if_all
+  end
```

Comment:
* [ ] should have working `my_all?`  identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_all? 

- [ ] when no block or argument is given returns true only when none of the collection members are false or nil, if the mentioned condition is not met it, should return false

```ruby     
[nil, false, true, []].my_all? == [nil, false, true, []].all? => false
```
- Yours is giving LocalJumpError
![image](https://user-images.githubusercontent.com/52209067/81132439-a8f0ff80-8f0b-11ea-8522-2940601e4c26.png)

- Real all?
![image](https://user-images.githubusercontent.com/52209067/77688266-67148700-6f65-11ea-82e5-cbef16444817.png)

- [ ] when a class is passed as an argument returns true only if all of the collection is a member of such class, otherwise, it should return false

```ruby     
[1, 2.5, 'a', 9].my_all?(Integer) == [1, 2.5, 'a', 9].all?(Integer) => false
```
- Yours is retrieving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/81132487-cfaf3600-8f0b-11ea-948d-9d310979ad55.png)

- Real all?
![image](https://user-images.githubusercontent.com/52209067/77688467-be1a5c00-6f65-11ea-8970-15c9e4f4913f.png)

- [ ] when a Regex is passed as an argument returns true if all of the collection matches the Regex

```ruby     
%w[dog door rod blade].my_all?(/d/) == %w[dog door rod blade].all?(/d/) => true
```
- Yours is retrieving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/81132506-e786ba00-8f0b-11ea-96b7-60e70ed09a1c.png)

- Real all?
![image](https://user-images.githubusercontent.com/52209067/77688741-31bc6900-6f66-11ea-90ec-b6a7cee7e6ca.png)

- [ ] when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern

```ruby     
[3,4,7,11].my_all?(3) == [3,4,7,11].all?(3)  => false 
```
- Yours is retrieving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/81132533-01c09800-8f0c-11ea-813a-fed5c6b44f93.png)

- Real all?
![image](https://user-images.githubusercontent.com/52209067/77689001-924ba600-6f66-11ea-9b72-210b3bb1d78b.png)